src/org/wikipedia/Wiki.java

```diff
@@ -4919,8 +4918,8 @@ public boolean isWatched(String title) throws IOException, CredentialNotFoundExc
                 boolean iscat = namespace(member) == CATEGORY_NAMESPACE;
                 if (maxdepth > 0 && iscat && !visitedcategories.contains(member))
                 {
-                    String[] categoryMembers = getCategoryMembers(member, --maxdepth, visitedcategories, ns);
                     visitedcategories.add(member);
+                    String[] categoryMembers = getCategoryMembers(member, --maxdepth, visitedcategories, ns);
```

Comment:
No, the meaning is the same in C++: decrement maxdepth by one, then pass it to the method. You want to leave maxdepth untouched in this case because there may be more than one subcategory at each level.